### PR TITLE
[8.x] Bug-fix and performance improvements in "TaggedCache" for "getMultiple" and "many" methods

### PR DESCRIPTION
**Pull request by yahya-uddin**
- This fixes the issue in: https://github.com/laravel/framework/issues/36778

- Improved documentation of `many` method in `Repository`

- Improves performance of `getMultiple` and `many` for `TaggedCache`. The method no longer calls the cache store with multiple `get` calls, but with one `many` call.

- Fixes repository so that it uses the `itemKey` method for `many`, `increment` and `decrement`, so it's consistent with other methods like `get` and `put`. This also makes a lot of the code in `TaggedCache` redundant.

### CODE REVIEW

**Comment by driesvints** [comment]
Although I'm not 100% sure about the implementation I can confirm this fixes the bug with `getMultiple`.

But I don't think we can make the `itemKey` changes in the same PR. If people were applying the `itemKey` in their code it could lead to unexpected behavior. I think it's best that that part is sent separately to master.

File: src/Illuminate/Cache/Repository.php [hunk 1]
```diff
@@ -115,8 +115,7 @@ public function get($key, $default = null)
      *
      * Items not found in the cache will use the provided default value, or `null` if not specified.
      *
-     * @param  array  $keys A list of keys that can obtained in a single operation. To specify a default value, use the
```
Comment (anik786) [inline src/Illuminate/Cache/Repository.php]: What was wrong with the original documentation. It more accurately represents what the function does.

**Comment by anik786** [comment]
> Although I'm not 100% sure about the implementation I can confirm this fixes the bug with `getMultiple`.
> 
> But I don't think we can make the `itemKey` changes in the same PR. If people were applying the `itemKey` in their code it could lead to unexpected behavior. I think it's best that that part is sent separately to master.

What part of the implementation did you think needs improving?
Also should I make a commit removing `itemKey` in `master` in that case, and take it out of this PR?

File: src/Illuminate/Cache/Repository.php (hunk 1 above)
Comment (driesvints) [reply src/Illuminate/Cache/Repository.php]: We don't document parameters individually anywhere in our open source projects.

**Comment by driesvints** [comment]
@anik786 the code feels a little bloated but I don't know how to improve it myself.

And yes the itemKey changes should be removed from this PR and sent in separately to master.

**Comment by taylorotwell** [comment]
Fixed the underlying bug here (https://github.com/laravel/framework/commit/0d211947da9ad222fa8eb092889bb7d7f5fb1726) without the other changes since those should really be separate.

**Pull request closed without merging**